src/commands/Misc/ship.ts

```diff
@@ -0,0 +1,93 @@
+import { SkyraCommand } from '@lib/structures/SkyraCommand';
+import { CanvasColors } from '@lib/types/constants/Constants';
+import { UserSettings } from '@lib/types/settings/UserSettings';
+import { socialFolder } from '@utils/constants';
+import { fetchAvatar, loadImage, roundNumber } from '@utils/util';
+import { Image } from 'canvas';
+import { Canvas } from 'canvas-constructor';
+import { remove as removeConfusables } from 'confusables';
+import { User } from 'discord.js';
+import { CommandStore, KlasaMessage } from 'klasa';
+import { join } from 'path';
+
+export default class extends SkyraCommand {
+
+	private removeSymbolsRegex = /^(?:[~`!@#%^&*(){}[\];:"'<,.>?/\\|_+=-])+$/g;
+	private lightThemeTemplate: Buffer | null = null;
+	private darkThemeTemplate: Buffer | null = null;
+	private heartIcon: Image | null = null;
+
+	public constructor(store: CommandStore, file: string[], directory: string) {
+		super(store, file, directory, {
+			cooldown: 10,
+			description: language => language.tget('COMMAND_SHIP_DESCRIPTION'),
+			extendedHelp: language => language.tget('COMMAND_SHIP_EXTENDED'),
+			requiredPermissions: ['ATTACH_FILES'],
+			runIn: ['text'],
+			usage: '(firstUser:firstUser) (secondUser:secondUser)',
+			usageDelim: ' '
+		});
+
+		this.createCustomResolver('firstUser', (arg, possible, message) => {
+			if (!arg) return message.guild?.members.random()!.user;
+			return this.userNameArgument.run(arg, possible, message);
+		});
+
+		this.createCustomResolver('secondUser', (arg, possible, message) => {
+			if (!arg) return message.guild?.members.random()!.user;
+			return this.userNameArgument.run(arg, possible, message);
+		});
+	}
+
+	private get userNameArgument() {
+		return this.client.arguments.get('username')!;
+	}
+
+	public async run(message: KlasaMessage, [firstUser, secondUser]: [User, User]) {
+		const [avatarFirstUser, avatarSecondUser] = await Promise.all([
+			fetchAvatar(firstUser, 64),
+			fetchAvatar(secondUser, 64),
+			message.author.settings.sync()
+		]);
+
+		const darkTheme = message.author.settings.get(UserSettings.DarkTheme);
+		const shipName = removeConfusables(
+			firstUser.username.substring(0, roundNumber(firstUser.username.length / roundNumber((Math.random() * 4) + 2))) +
+			secondUser.username.substring(roundNumber(secondUser.username.length / roundNumber((Math.random() * 4) + 2)))
+		).replace(this.removeSymbolsRegex, '');
+
+		const DATA = message.language.tget('COMMAND_SHIP_DATA');
+		const attachment = await new Canvas(224, 88)
+			.save()
+			.addImage(darkTheme ? this.darkThemeTemplate! : this.lightThemeTemplate!, 0, 0, 224, 88)
```

Comment:
You could simplify this by removing the `!` from the end, and where you define the buffers:
```ts
private darkThemeTemplate!: Buffer;
```